velox/dwio/dwrf/writer/ColumnWriter.cpp

```diff
@@ -795,7 +797,9 @@ class StringColumnWriter : public BaseColumnWriter {
         dictEncoder_{
             getMemoryPool(MemoryUsageCategory::DICTIONARY),
             getMemoryPool(MemoryUsageCategory::GENERAL)},
-        rows_{getMemoryPool(MemoryUsageCategory::GENERAL)},
+        rows_{
+            getMemoryPool(MemoryUsageCategory::GENERAL),
```

Comment:
Memory pool does quantized memory reservation so is there any difference for 16 vs 1024 items? Thanks!

Reply:
@xiaoxmeng, this happens in the same memory pool 20k-30k times and it really adds up because each allocation is 1024 uint32_t types. I assume that the quantization happens at the pool level and not every allocation, then it's about the total amount of memory that has to be allocated.

In the above case it's 80MB-120MB per such column. Let me know if that's the right understanding of quantized memory allocation/accounting.